### ISSUE TITLE
[Bug/v2]: ComboboxItem unhandled error in mounted

### ISSUE THREAD

**Issue by RayGuo-ergou** [radix-vue/radix-vue#1234]
### Environment

```bash
Developement/Production OS: Windows 11
Node version: 20
Package manager: pnpm@8.6.0
Reka Vue version: 0.0.2
Vue version: ^3.0.0
Client OS: Windows 11
Browser: Firefox
```

### Link to minimal reproduction

https://github.com/RayGuo-ergou/reka-reproduction

### Steps to reproduce

Clone the repo, run the dev server, then in app search anything, delete all text then search again that will make the `computed` as empty array.

```ts
const people = [
  { id: 1, name: 'Durward Reynolds'},
  { id: 2, name: 'Kenton Towne'},
  { id: 3, name: 'Therese Wunsch'},
  { id: 4, name: 'Benedict Kessler'},
  { id: 5, name: 'Katelyn Rohan'},
]
const selectedPeople = ref(people[0])
const search = ref('')

const searchResult = computed(()=>{
  return people.filter(i => !i.name.startsWith(search.value))
})
```

<details><summary>Warning</summary>
<p>

[Vue warn]: Unhandled error during execution of mounted hook 
  at <ComboboxItem key=1 value= 
Object { id: 1, name: "Durward Reynolds" }
 > 
....
</p>
</details> 

<details><summary>Error</summary>
<p>

Uncaught (in promise) TypeError: can't access property "textContent", u.value is undefined

</p>
</details> 

### Describe the bug

I was trying v2 aka reka with shadcn-ui, and I found that a breaking changing in combobox may break the [tags-input](https://www.shadcn-vue.com/docs/components/tags-input.html)

In https://github.com/radix-vue/radix-vue/commit/ee8a3f2366a5c27c2bf1cc0a1ecbb0fea559a9f7#diff-19766b3de8934964a0c2730b5e4d28c90759d1b67572135ba3be029752b4d31dR50

The `currentElement` will be undefined if the list of items become an empty array.

### Context & Screenshots (if applicable)

![GIF 19-08-2024 12-35-05 PM](https://github.com/user-attachments/assets/66a0dcb6-d58d-460f-a0c0-f9775dec445e)

**Comment by zernonia** [comment]
@RayGuo-ergou  v2 has breaking changes, and migration for `shadcn-vue` will not be so direct. So can I close this ticket off  as it's not a bug but something needs to be updated.

**Comment by RayGuo-ergou** [comment]
> So can I close this ticket off as it's not a bug but something needs to be updated.

Yeah sure thing, but just want to ask "need to be updated" do you mean reka side or shadcn side?

**Comment by RayGuo-ergou** [comment]
I have tried my reproduction again weiredly

```ts
const searchResult = computed(()=>{
  return people.filter(i => i.name.startsWith(search.value))
})
```

This will not trigger this issue but

```ts
const searchResult = computed(()=>{
  return people.filter(i => !i.name.startsWith(search.value))
})
```
This will

**Comment by RayGuo-ergou** [comment]
Never mind ^, both indeed have this issue, and my reproduction is based on https://github.com/radix-vue/radix-vue/blob/ee8a3f2366a5c27c2bf1cc0a1ecbb0fea559a9f7/packages/radix-vue/src/Combobox/story/_ComboboxManualFilter.vue

except I am using native javascript `startsWith`

So I would say this is a bug.

**Comment by RayGuo-ergou** [comment]
```ts
        <template v-if="!!searchResult.length">
          <template v-for="person in searchResult" :key="person.id">
            <ComboboxItem
              :value="person"
            >
              <span>
                {{ person.name }}
              </span>
            </ComboboxItem>

          </template>
        </template>
```

This way seems not trigger it (since it's re-rendering i guess?)

I am not sure again 🥲, well I just played with v2 today.

I will leave it to your for deciding to keep this issue open or not.😁

**Comment by RayGuo-ergou** [comment]
Hi @zernonia Sorry for pining, but I guess you only have notifications for pined comment?

I have came to a conclusion that this is a bug with combobox in v2, for example I can even trigger this using example from the doc. 

Here's an screenshot from the story in v2.
![image](https://github.com/user-attachments/assets/45390d64-3c79-4d16-843b-5c3f0d7f5d76)

Sorry for not describe this issue clearly from beginning, and I understand v2 in very early stage so my intention is mostly let you know this issue.

I am also happy to look into this sometime next month :-).

**Comment by zernonia** [comment]
@RayGuo-ergou  can you check again with [reka alpha release](https://github.com/unovue/radix-vue/releases/tag/v1.0.0-alpha.1)

**Comment by RayGuo-ergou** [comment]
Hi @zernonia thanks for asking, no with alpha release it still have the same issue. Same for the example from the store, and it's very weird that initially works fine then after few random actions (e.g. input, remove input and refresh the page) it then always throws the error above.

Apologies to make a promise but did not do it last month. I took some days just resting, will try to look into it this weekend.

**Comment by zernonia** [comment]
Can you please share the reproduction using this stackblitz? https://stackblitz.com/edit/hn2yox?file=src%2FApp.vue

**Comment by RayGuo-ergou** [comment]
The items has to be fetched from a server to reproduce this.
Here's the link: https://stackblitz.com/edit/hn2yox-yy4md2?file=src%2FApp.vue

For me after the data fetched I will see this. 

![image](https://github.com/user-attachments/assets/a9099e09-a8ba-401c-8e5d-ee1a3a4febd5)

![image](https://github.com/user-attachments/assets/d9421c41-2c93-4bfc-b6aa-76819e2c49a1)

**Comment by RayGuo-ergou** [comment]
Hi @zernonia 

The patch that seems to fix this:

```diff
diff --git a/packages/core/src/Combobox/ComboboxItem.vue b/packages/core/src/Combobox/ComboboxItem.vue
index a03aec8b..a59f5e4a 100644
--- a/packages/core/src/Combobox/ComboboxItem.vue
+++ b/packages/core/src/Combobox/ComboboxItem.vue
@@ -44,10 +44,16 @@ if (!props.value) {
 }
 
 const isRender = computed(() => {
-  if (rootContext.isVirtual.value || rootContext.ignoreFilter.value || !rootContext.filterState.search)
+  if (rootContext.isVirtual.value || rootContext.ignoreFilter.value || !rootContext.filterState.search) {
     return true
-  else
-    return rootContext.filterState.filtered.items.get(id)! > 0
+  }
+  else {
+    const filteredCurrentItem = rootContext.filterState.filtered.items.get(id)
+    if (filteredCurrentItem === undefined) {
+      return true
+    }
+    return filteredCurrentItem > 0
+  }
 })
 
 onMounted(() => {
```
<hr/>
Because in case of async fetching data to use for `comboboxItems`, `isRender` is false for the new items(will explain below). And then after getting the data from backend, as `onMounted` call already finished. So `rootContext.allItems.value.set` will not execute when `isRender` become `true`. These new "items" are lost.

### Why the error appear

The happens before what happened above.

```ts
  if (rootContext.isVirtual.value || rootContext.ignoreFilter.value || !rootContext.filterState.search) {
```

^ This is false in this case. So it then goes to `else` block and returns `false` for `isRender` because `undefined > 0 === false`

Thus, `ListboxItem` is not rendered, which makes `currentElement` is undefined. So it triggers the error shows above.

I am not sure if this is the approach/correct-fix so I leave as a comment rather than PR. Let me know if you are happy for this to be a PR :).

**Comment by RayGuo-ergou** [comment]
Having a second thought, this bug is not limited to async render. It applies to all items if `if (rootContext.isVirtual.value || rootContext.ignoreFilter.value || !rootContext.filterState.search) {` is false.

**Comment by sunnylost** [comment]
This bug made me realize that the `currentElement` obtained from `usePrimitiveElement()` could be `undefined`, so optional chaining should be used instead of directly accessing the property.

**Comment by RayGuo-ergou** [comment]
> This bug made me realize that the currentElement obtained from usePrimitiveElement() could be undefined, so optional chaining should be used instead of directly accessing the property.

Right, at the moment `currentElement` is declared as `ref<HtmlElement>`. Perhaps should change to `ref<undefined|HtmlElement>` fisrt?
